File: Dsa-day151.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        int n = prices.length;
        if (n == 0) return 0;
        long ans = 1;
        int length = 1;
        for (int i = 1; i < n; ++i) {
            if (prices[i] == prices[i - 1] - 1) {
                ++length;
            } else {
                length = 1;
            }
            ans += length;
        }
        return ans;
    }
}

File: Dsa-day151.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        n = len(prices)
        if n == 0:
            return 0
        ans = 1
        length = 1
        for i in range(1, n):
            if prices[i] == prices[i - 1] - 1:
                length += 1
            else:
                length = 1
            ans += length
        return ans